src/main.rs

```diff
@@ -53,24 +63,45 @@ fn main() -> Result<()> {
     };
 
     if let Some(Commands::Stats { all }) = cli.command {
+        if cli.query.is_some() {
+            // todo implement --query for stats
+            error!("Sorry, '--query' is not implemented for stats yet.");
+        }
         info!("Getting stats");
         stats::write(
             &util::load_relations(
                 cli.in_file,
-                if all {
-                    util::filter_all_relations
-                } else {
-                    util::filter_target_relations
-                },
+                if all { filter::all } else { filter::by_target },
             )?,
             out,
         )?;
     } else {
         info!("Extracting localities");
-        locality::write(
-            &util::load_relations(cli.in_file, util::filter_target_relations)?,
-            out,
-        )?;
+
+        let filter = {
+            let query_filter = cli.query.as_ref().map(|query| filter::by_query(query));
+
+            move |obj: &OsmObj| -> bool {
+                filter::by_target(obj) && query_filter.as_ref().map_or(true, |f| f(obj))
+            }
```

Comment:
Ich finde `move` closures immer trickier als non-`move` closures und würde persönlich `query_filter` in der closure anlegen. Das klappt, falls du direkt direkt `filter::by_query` über `cli.query` mapst. Das braucht zwar ein explizites `as_deref`, was fast noch ekliger ist als `as_ref`, aber ist dafür hier kürzer 🤷 (eklig weil der compiler schon oft automatische derefs einfügt und `x.as_ref()` einfach `&x` ist). 

```suggestion
        let filter = |obj: &OsmObj| -> bool {
            let query_filter = cli.query.as_deref().map(filter::by_query);
            filter::by_target(obj) && query_filter.map_or(true, |f| f(&obj))
        };
```

Reply:
Ich finde es eigtl. besser, den query_filter zuerst für alle Durchläufe zu erstellen

Reply:
Falls ich nichts verpasse macht meine suggestion genau das gleiche, nur anders geschrieben.

Reply:
stimmt